modules/@apostrophecms/permission/index.js

```diff
@@ -61,7 +61,11 @@ module.exports = {
         }
         const type = docOrType && (docOrType.type || docOrType);
         const doc = (docOrType && docOrType._id) ? docOrType : null;
-        const manager = type && self.apos.doc.getManager(type);
+        let manager = type && self.apos.doc.getManager(type);
+        if (doc && !manager) {
+          // Fallback to an appropriate manager module
+          manager = self.apos.util.getManagerOf(doc);
```

Comment:
We don't support viewRole etc. for widget types. This shouldn't be necessary. We should not be calling permission.can with a widget ever actually. Are we doing that though?

Reply:
yes, in the annotate function:

```js
      annotate(req, action, objects) {
        const property = `_${action}`;
        for (const object of objects) {
          if (self.can(req, action, object)) {
            object[property] = true;
          }
        }
      },
```

we could do something like that to exclude widgets, WDYT?

```js
      annotate(req, action, objects) {
        const property = `_${action}`;
        for (const object of objects) {
          if (object.metaType !== 'widget' && self.can(req, action, object)) {
            object[property] = true;
          }
        }
      },
```

Reply:
I think we would have to check that the `metaType` is `doc`, to exclude other things that won't work, but yes I agree with this approach.